projects/lights_control/test/test_lights_signal.c

```diff
@@ -0,0 +1,133 @@
+// This tests the signal FSM and makes sure the state transitions are properly being done.
+// NOTE: The assertions work based on the raised event immediately after turning on a blinker.
+// It is assumed that the soft-timer interrupt is not being triggered before the event assertion.
+
+#include "lights_blinker.h"
+#include "lights_events.h"
+#include "lights_signal_fsm.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+#define TEST_LIGHTS_CAN_ASSERT_RAISED_EVENT(event, peripheral) \
+  do {                                                         \
+    const Event raised_event = { 0 };                          \
+    TEST_ASSERT_OK(event_process(&raised_event));              \
+    TEST_ASSERT_EQUAL(peripheral, raised_event.data);          \
+    TEST_ASSERT_EQUAL(event, raised_event.id);                 \
+  } while (0)
+
+#define TEST_LIGHTS_CAN_ASSERT_NO_EVENT_RAISED()      \
+  do {                                                \
+    const Event raised_event = { 0 };                 \
+    TEST_ASSERT_NOT_OK(event_process(&raised_event)); \
+  } while (0)
+
+typedef enum {
+  TEST_LIGHTS_SIGNAL_FSM_CMD_OFF = 0,
+  TEST_LIGHTS_SIGNAL_FSM_CMD_ON,
+  NUM_TEST_LIGHTS_SIGNAL_FSM_CMDS
+} TestLightsSignalFsmCmd;
+
+static LightsBlinkerDuration s_duration_ms = 300;
+
+static LightsSignalFsm s_signal_fsm = { 0 };
+
+void setup_test(void) {
+  event_queue_init();
```

Comment:
Don't you need soft timers etc. for blinker to work?

Reply:
Yeah, this was using the mock, which didn't use timers... I've merged with master and replaced mocks with actual `lights_blinker` and it complained and I fixed it.